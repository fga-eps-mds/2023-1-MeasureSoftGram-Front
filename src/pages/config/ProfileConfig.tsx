import useRequireAuth from '@hooks/useRequireAuth';
import { NextPageWithLayout } from '@pages/_app.next';
import Head from 'next/head';
import { getLayout } from '@components/Layout/getLayout';
import { Container, TextField, InputAdornment, IconButton } from '@mui/material';
import { Visibility, VisibilityOff } from '@mui/icons-material';
import { useEffect, useState } from 'react';
import { getAccessToken } from '@services/Auth';

const ProfileConfig: NextPageWithLayout = () => {
  useRequireAuth();

  const [tokenVisible, setTokenVisible] = useState(false);
  const [apiToken, setApiToken] = useState('');

  useEffect(() => {
    getAccessToken().then((res) => {
      if (res.type === 'success') {
        setApiToken(res.value.key);
      }
    });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const handleToggleVisibility = () => {
    setTokenVisible(!tokenVisible);
  };

  return (
    <>
      <Head>
        <title>Config</title>
      </Head>
      <Container>
        <h1>Configurações</h1>

        <h3>Chave secreta para acesso à API</h3>
        <TextField
          type={tokenVisible ? 'text' : 'password'}
          value={apiToken}
          label="Chave secreta"
          InputProps={{
            endAdornment: (
              <InputAdornment position="end">
                <IconButton aria-label="toggle password visibility" onClick={handleToggleVisibility}>
                  {tokenVisible ? <VisibilityOff /> : <Visibility />}
                </IconButton>
              </InputAdornment>
            )
          }}
        />
      </Container>
    </>
  );
};

ProfileConfig.getLayout = getLayout;

export default ProfileConfig;
