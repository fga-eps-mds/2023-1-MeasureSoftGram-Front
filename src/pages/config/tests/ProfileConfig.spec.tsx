import { render } from '@testing-library/react';
import ProfileConfig from '../ProfileConfig';

describe('ProfileConfig', () => {
  it('should render successfully', () => {
    const { container } = render(<ProfileConfig />);
    expect(container).toMatchSnapshot();
  });
});
