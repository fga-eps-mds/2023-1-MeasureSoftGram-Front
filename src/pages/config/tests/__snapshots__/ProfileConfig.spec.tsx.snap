// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileConfig should render successfully 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <h1>
      Configurações
    </h1>
    <h3>
      Chave secreta para acesso à API
    </h3>
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for=":r0:"
        id=":r0:-label"
      >
        Chave secreta
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r0:"
          type="password"
          value="YourAPIToken"
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
        >
          <button
            aria-label="toggle password visibility"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="VisibilityIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-14lo706"
          >
            <span>
              Chave secreta
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
